scripts/project.js

```diff
@@ -1,44 +1,57 @@
-function Project(opts){
-  this.title = opts.title;
-  this.projectUrl = opts.projectUrl;
-  this.publishedOn = opts.publishedOn;
-  this.projectDescription = opts.projectDescription;
-}
-
-Project.all = [];
-
-Project.prototype.toHtml = function() {
-  var source = $('#project-template').html();
-  var template = Handlebars.compile(source);
-  return template(this);
-}
-
-Project.loadAll = function(rawData) {
-  rawData.forEach(function(ele) {
-    Project.all.push(new Project(ele));
-  })
-};
-
-Project.fetchAll = function() {
-  var storedEtag;
-  $.ajax({
-    type: "HEAD",
-    url: "data/portfolioProjects.json",
-    success: function(data, message, xhr) {
-      storedEtag = xhr.getResponseHeader("etag");
-      if (storedEtag === localStorage.etag && localStorage.rawData) {
-        Project.loadAll(JSON.parse(localStorage.rawData));
-        projectView.initIndexPage();
-      } else {
-        $.getJSON('data/portfolioProjects.json', function(data) {
-          console.log(data);
-          Project.loadAll(data);
-          localStorage.setItem("etag", JSON.stringify(storedEtag));
-          localStorage.setItem('rawData', JSON.stringify(data));
+(function(module){
+  function Project(opts){
+    Object.keys(opts).forEach(function(e, index, keys) {
+      this[e]= opts[e];
+    },this);
+  }
+
+  Project.all = [];
+
+  Project.prototype.toHtml = function(type) {
+    var source = $('#' + type +'-template').html();
+    var template = Handlebars.compile(source);
+    return template(this);
+  }
+
+  Project.loadAll = function(rawData){
+    Project.all = rawData.map(function(ele) {
+      return new Project(ele);
+    });
+  };
+
+  Project.fetchAll = function() {
+    var storedEtag;
+    $.ajax({
+      type: "HEAD",
+      url: "data/portfolioProjects.json",
+      success: function(data, message, xhr) {
+        storedEtag = xhr.getResponseHeader("etag");
+        if (storedEtag === localStorage.etag && localStorage.rawData) {
+          Project.loadAll(JSON.parse(localStorage.rawData));
           projectView.initIndexPage();
-          });
+        } else {
+          $.getJSON('data/portfolioProjects.json', function(data) {
+            Project.loadAll(data);
+            console.log(data);
+            localStorage.setItem("etag", JSON.stringify(storedEtag));
+            localStorage.setItem('rawData', JSON.stringify(data));
+            projectView.initIndexPage();
+            });
+        }
       }
-    }
-  });
+    });
+
+  };
+
+  Project.prototype.findDescWordCount = function(){
+    return Project.all.map(function(project){
+      return project.projectDescription.split(/\b\S+\b/g).length
+    })
+    .reduce(function(prevTotal, curr){
+      return Project.prototype.wordCount = prevTotal + curr;
```

Comment:
I'm not sure why you're using Project.prototype.wordCount here. If it's on the prototype chain, you should be just able to use (instance of Project object).wordCount. Read Jeff's link in the Slack about Javascript and prototype, because this is really important to understand.